content/post/094-js-ipfs-0.44.md

```diff
@@ -0,0 +1,177 @@
+const controller = new AbortController()
+
+setTimeout(() => {
+  controller.abort()
+}, 1000)
+
+try {
+  await ipfs.get(cid, {
+    signal: controller.signal
+  })
+} catch (err) {
+  console.err(err)
+}
+```
+
+The AbortSignal is passed all the way down the call stack to the libp2p and ipld components that underpin IPFS. In this initial release, timeouts are supported by the IPFS layer and in future releases libp2p and ipld will use the AbortSignal to also free up resources and perform any other necessary cleanup that would have been done if the request had completed successfully.
+
+## New browser datastore
+
+`js-IPFS@0.44.0` brings a new datastore to the browser. In the browser, all blocks and other repo data is currently stored in [IndexedDB](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API), given that it is the only way available to persist and query large amounts of user data at the time of writing.
+
+In node and go-ipfs, [LevelDB](https://en.wikipedia.org/wiki/LevelDB) has been the datastore of choice for application data (though blocks have been stored on the filesystem) and go-IPFS is moving to [Badger](https://dgraph.io/blog/post/badger/)).
+
+Out of this grew [interface-datastore](https://github.com/ipfs/interface-datastore)—a spec for storing key/value pairs implemented by the datastores used by IPFS. In the browser this was [datastore-level](https://github.com/ipfs/js-datastore-level) which was backed by [level-js](https://www.npmjs.com/package/level-js), which was backed by IndexedDB.
+
+Phew.
+
+To remove a few layers here, we’ve created the [datastore-idb](https://github.com/ipfs/js-datastore-idb) which implemenets the interface-datastore spec, backed by IndexedDB without going via level.
+
+What does this mean?
+
+The new datastore is smaller and faster and the upgrade is seamless since it’s still IndexedDB under the hood:
+
+```
+datastore-idb up to 44% faster and 69% smaller
+
+batch idb x 5.26 ops/sec ±4.57% (29 runs sampled)
+batch level x 2.92 ops/sec ±3.76% (19 runs sampled)
+Fastest is batch idb
+
+size 37.32KB to 11.41KB
+```
+
+## ✨New features
+
+* Cancellable API calls ([#2993](https://github.com/ipfs/js-ipfs/issues/2993)) ([2b24f59](https://github.com/ipfs/js-ipfs/commit/2b24f590041a0df9da87b75ae2344232fe22fe3a)), closes [#3015](https://github.com/ipfs/js-ipfs/issues/3015)
+* `ipfs.block.put` now accepts the `pin` argument in line with go-ipfs [#3015](https://github.com/ipfs/js-ipfs/issues/3015)
+
+## 🦟 Bugs fixed
+
+* Fixes browser script tag example ([#3034](https://github.com/ipfs/js-ipfs/issues/3034)) ([ee8b769](https://github.com/ipfs/js-ipfs/commit/ee8b769b96f7e3c8414bbf85853ab4e21e8fd11c)), closes [#3027](https://github.com/ipfs/js-ipfs/issues/3027)
+* No longer produces browser bundle with all IPLD formats in the browser ([#3025](https://github.com/ipfs/js-ipfs/issues/3025)) ([e6079c1](https://github.com/ipfs/js-ipfs/commit/e6079c17d5656e92dd5191f0581000c6a782c7ed))
```

Comment:
Does this reduce bundle size? should mention, if so

Reply:
No, it's just an extra file that wasn't used.